src/extension/src/TelemetryWriter.py

```diff
@@ -13,75 +13,107 @@
-
-    def send_debug_info(self, error_info):
-        # Usually expected to instrument possibly problematic code
-        return self.try_send_message(error_info, Constants.TelemetryDebug)
-
-    def send_info(self, info):
-        # Usually expected to be significant runbook output
-        return self.try_send_message(info, Constants.TelemetryInfo)
-    # endregion
-
-    # Composed payload
-    def send_machine_config_info(self):
-        # Machine info
-        machine_info = {
-            # The commented out code needs to be addressed similar to how Core does it for Python 3.8+ support.
-            # Not an issue at the time of this writing as telemetry support has not been implemented.
-            #  'platform_name': str(platform.linux_distribution()[0]),
-            #  'platform_version': str(platform.linux_distribution()[1]),
-            'machine_arch': str(platform.machine())
-        }
-        return self.send_config_info(machine_info, 'machine_config')
-
-    def send_execution_error(self, cmd, code, output):
-        # Expected to log any errors from a cmd execution, including package manager execution errors
-        error_payload = {
-            'cmd': str(cmd),
-            'code': str(code),
-            'output': str(output)[0:3072]
+        self.events_folder_path = None
+        self.operation_id = ""
+
+    def __new_event_json(self, task_name, event_level, message):
+        return {
+            "Version": Constants.EXT_VERSION,
+            "Timestamp": str((datetime.datetime.utcnow()).strftime("%Y-%m-%dT%H:%M:%SZ")),
+            "TaskName": task_name,
+            "EventLevel": event_level,
+            "Message": self.__ensure_message_restriction_compliance(message),
+            "EventPid": "",
+            "EventTid": "",
+            "OperationId": self.operation_id  # we can provide activity id from config settings here, but currently we only read settings file for enable command
         }
-        return self.send_error_info(error_payload)
-    # endregion
 
-    # region Transport layer
-    def try_send_message(self, message, category=Constants.TelemetryInfo):
-        raise NotImplementedError
+    @staticmethod
+    def __ensure_message_restriction_compliance(full_message):
+        """ Removes line breaks, tabs and restricts message to a character limit """
+        message_size_limit = Constants.TELEMETRY_MSG_SIZE_LIMIT_IN_CHARACTERS
+        formatted_message = re.sub(r"\s+", " ", str(full_message))
+        return formatted_message[:message_size_limit - 3] + '...' if len(formatted_message) > message_size_limit else formatted_message
+
+    def write_event(self, task_name, message, event_level=Constants.TelemetryEventLevel.Informational):
+        # create event json
+        # check if event size complies with restriction i.e. 6k
+        # write event to events file
+        if os.path.getsize(self.events_folder_path) >= Constants.TELEMETRY_DIR_SIZE_LIMIT_IN_CHARACTERS:
```

Comment:
Please sort out the characters vs. bytes issue, in actual outcomes, then in implementation and backwards to the variable names used.